Разметка главной страницы

AlexandraPoltoratskaya opened the pull request:
---
:mortar_board: [Разметка главной страницы](https://up.htmlacademy.ru/htmlcss/33/user/232847/tasks/3)

AlexandraPoltoratskaya committed:
Update Readme.md

Внесла ФИО наставника

AlexandraPoltoratskaya committed:
выделила блоки и смысловые разделы 1й страницы Device

AlexandraPoltoratskaya committed:
сверстала хедер сайта и слайдер

AlexandraPoltoratskaya committed:
Изменила мета тег тайтл на более маркетинговый

AlexandraPoltoratskaya committed:
сверстала иконки каталога на 1й странице Девайс

AlexandraPoltoratskaya committed:
завершила верстку 1й страницы Device

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>Device | Все для Метавселенной | Цукерберг обзавидуется</title>
```
метавселенная и цукерберг это, конечно, увлекательно, но всё-таки к магазину инструментов ни то, ни другое отношения не имеет =]

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+      <a href="#"><img class="logo-header" src="" alt="Логотип магазина Девайс"></a>
+
+      <nav class="user-navigation">
+        <input type="search" name="Поиск по сайту" placeholder="Текст поиска">
+        <button>Найти</button>
```
поля для сбора / отправки данных должны находиться внутри тега form https://up.htmlacademy.ru/htmlcss/33/criteries#HTML-04 

элементам, сделанным тегом button лучше явно указывать атрибут type. у кнопок отправки формы значение атрибута type равно submit (и это также дефолтное значение атрибута). у остальных кнопок, которые никакую форму не отправляют, type="button".

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+          
+          <li class="main-navigation-item">
+            <a class="main-navigation-link" href="#">Каталог товаров</a>
+            <!--это решение скоммунижжено вот отсюда https://html5css.ru/howto/howto_js_dropdown.php-->
```
ну оно недокомунижжено, поэтому не работает

на статичных сайтах, написанных руками без использования библиотек хорошим тоном считается разделять разметку, стили и скрипты. инлайновые стили (кроме отдельных случаев вроде вот такого 
![HTML 1 : Девайс (Copy) 2022-01-23 23-38-52](https://user-images.githubusercontent.com/11165551/150697147-cff515e5-a43d-4ff8-a828-266add5d27a9.jpg)
тут положение бегунков логично писать в инлайновые стили, потому что их движение будет управляться через js, а js записывает стили элементам именно в инлайн) и инлайновый js считаются плохим тоном. onclick это инлайновый js, вернее, шаг в его сторону. и, разумеется, чтоб всё работало, где-то на странице должна быть функция myFunction(), которая вызывается по клику. 

и напоминаю, что на этом интенсиве мы не будем изучать js, поэтому реализовывать механику от нас не требуется, достаточно разметки

но структура разметки тут норм. только надо реальные пункты добавить в меню, они показаны в разделе "стайлгайд" макета 
![image](https://user-images.githubusercontent.com/11165551/150697391-cc78d83e-9f47-40fc-aa07-fd9bce74d010.png)

моноподы это ссылка на нашу будущую страницу каталога

AlexandraPoltoratskaya replied:
я удалила кусок с Java Script из версии, отправленной на повторную правку. там его быть не должно. значит, я отправила на повторную проверку не ту версию

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+            <a class="main-navigation-link" href="#">Каталог товаров</a>
+            <!--это решение скоммунижжено вот отсюда https://html5css.ru/howto/howto_js_dropdown.php-->
+            <div class="dropdown">
+              <button onclick="myFunction()" class="dropbtn"><img src="" alt="Открыть список категорий">Плюс</button>
```
из соображений единообразия именования классов стоит dropbtn переименовать в dropdown-btn

также у тега button не бывает атрибута alt

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+        </p>
+        <a class="slider-description-link" href="#">Подробнее</a>
+
+        <ol class="slider-pagination">
+          <li class="slider-pagination-item">
+            <button class="slider-pagination-button" slider-pagination-button-current" type="radio">
+              <span class="visually-hidden">1 товар</span>
+            </button>
+          </li>
+          <li class="slider-pagination-item">
+            <button class="slider-pagination-button" slider-pagination-button" type="radio">
+              <span class="visually-hidden">2 товар</span>
+            </button>
+          </li>
+          <li class="slider-pagination-item">
+            <button class="slider-pagination-button" slider-pagination-button" type="radio">
+              <span class="visually-hidden">3 товар</span>
+            </button>
+          </li>
+        </ol>
```
пагинация не имеет отношения к описанию отдельного слайда, она одна на весь слайдер

AlexandraPoltoratskaya replied:
я удалила пагинацию, и теперь это выглядит так:

```
 <ul class="slider-gallery">
          <li class="slider-gallery-item">
            <button class="slider-gallery-button slider-gallery-button-current" type="radio">
              <span class="visually-hidden">1 товар</span>
            </button>
          </li>
          <li class="slider-gallery-item">
            <button class="slider-gallery-button slider-gallery-button" type="radio">
              <span class="visually-hidden">2 товар</span>
            </button>
          </li>
          <li class="slider-gallery-item">
            <button class="slider-gallery-button slider-gallery-button" type="radio">
              <span class="visually-hidden">3 товар</span>
            </button>
          </li>
        </ul>
```

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+    <main class="main-index">
+      <h1 class="visually-hidden">Девайс - интернет-магазин гаджетов</h1>
+
+      <section class="slider-image">
```
скорее весь слайдер должен быть секцией, ну или по крайней мере все эти секции с классами с префиксом  slider- должны иметь общую обёртку, объединяющую их логически и физически. ну то есть вот это вот всё - slider 
![HTML 1 : Девайс (Copy) 2022-01-23 23-51-09](https://user-images.githubusercontent.com/11165551/150697723-a871b7cb-93d6-4445-b191-8eeb08f9dd46.jpg)

и начинать надо именно с крупных блочных обёрток, а потом их детализировать внутри. сейчас запчасти слайдера валяются внутри main вперемешку с контентом других секций. собрать из таких запчастей единый блок слайдера будет затруднительно.

также немного о структуре: 
в слайдере 3 слайда, а не один. контент всех слайдов можно увидеть в стайлгайде 
![image](https://user-images.githubusercontent.com/11165551/150697595-10155e07-f2f3-48c1-9bb4-5dddb225af30.png)

собственно стандартная структура слайдера такова

```
slider 
    slides 
        slide 
        slide
        slide
    slider-controls
        arrows
        pagination
```

в каждом слайде есть картинка, текстовое описание и ссылка на раздел в каталоге. 
помимо слайдов в слайдере есть контролы (стрелки переключают слайды, их размечаем интерактивными тегами. пагинация теоретически может тоже переключать, а может быть просто индикатором активного слайда, её можно разметить как интерактивными, так и обычными тегами. практически у нас в стайлгайде показаны состояния при взаимодействии с кнопками пагинации, то есть в нашем случае она интерактивна )

можно открыть какой-нибудь популярный плагин для слайдера и посмотреть, как делают там. например вот https://swiperjs.com/demos#navigation
![Swiper Demos 2022-01-24 00-05-36](https://user-images.githubusercontent.com/11165551/150699454-c24e04bc-a3f4-4ec3-a3e8-e72fc2b833e4.jpg)

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+      <h1 class="visually-hidden">Девайс - интернет-магазин гаджетов</h1>
+
+      <section class="slider-image">
+        <h2 class="visually-hidden">Изображение товара на слайдере</h2>
+        <img class="slider-big-image" src="" alt="Товар во всей красе">
```
тут перебор с описаниями, картинке хватит alt, не думаю, что для картинки нужна секция с заголовком. и лучше, чтоб в alt примерно описывалось, что мы видим (типа "изображение товара Квадрокоптер с лазером"). всё-таки в первую очередь нам нужно информировать покупателя как можно лучше, развлечь его поэзией это уже сверхзадача

есть формальные требования к описаниям https://up.htmlacademy.ru/htmlcss/33/criteries#ALLY-01 
в частности, в конце просят ставить точку

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+      <section class="slider-image">
+        <h2 class="visually-hidden">Изображение товара на слайдере</h2>
+        <img class="slider-big-image" src="" alt="Товар во всей красе">
+        <button class="slider-button slider-prev" type="button">
+          <span class="visually-hidden">Предыдущее фото</span>
+        </button>
+        <button class="slider-button slider-next" type="button">
+          <span class="visually-hidden">Следующее фото</span>
+        </button>
```
кнопки переключения слайдов одни на весь слайдер. то есть в каждом слайде их дублировать не надо

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+      <section class="slider-description">
+        <h2 class="visually-hidden">Описание товара на слайдере</h2>
+
+        <p class="slogan">Порхает как бабочка,<br>
```
br пока лучше не добавлять, ближе к концу проекта можно будет их расставить там, где иначе переноса добиться невозможно. но это точно не тот случай, потому что первая строка длиннее второй, и перенос автоматически случится в нужном месте (слово "жалит") в первую строку просто не влезет уже 

а вот в этих двух случаях именно br, потому что заголовки нестандартной ширины, и на первой строке ещё много места
![HTML 1 : Девайс (Copy) 2022-01-24 01-00-49](https://user-images.githubusercontent.com/11165551/150699559-cb9cefd9-62e4-4a69-9cdb-4a5f0e30af32.jpg)

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+
+        <p class="slogan">Порхает как бабочка,<br>
+                          жалит как пчела!</p>
+        <p class="description">Этот обычный, на первый взгляд, квадрокоптер оснащен мощным лазером, замаскированным под стандартную камеру.
+        </p>
```
обрати внимание на вот этот раздел технических требований https://up.htmlacademy.ru/htmlcss/33/criteries#PROJ-01 и собственно на кодгайд http://codeguide.academy/html-css.html
http://codeguide.academy/html-css.html#html-tags-formatting тут сказано, что текст должен быть на одной строке с тегом. то есть перед закрывающим тегом `</p>` перенос строки не нужен

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+
+        <ol class="slider-pagination">
+          <li class="slider-pagination-item">
+            <button class="slider-pagination-button" slider-pagination-button-current" type="radio">
```
у тега button не бывает значения type, равного radio. тут просто button type="button"

и тут какой-то неведомый атрибут slider-pagination-button-current объявился, к которому прилипла кавычка. видимо результат неудачной копипасты

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+        </ol>
+      
+        <ul class="slider-list">
+          <li class="slider-list-item">
+            <b class="slider-list-item-number">800 м</b>
+            <span class="slider-list-item-name">Дальность полета</span>
+            </li>
```
нужно следить за форматированием кода. сейчас закрывающий тег li визуально находится на одном уровне с лежащим в нём тегом span, а должен быть сдвинут относительно него влево на размер отступа (собственно на тот же уровень, на котором открывается этот тег li) 
![Разметка главной страницы by AlexandraPoltoratskaya · Pull Request #1 · htmlacademy-htmlcss:232847-device-33 2022-01-24 01-07-44](https://user-images.githubusercontent.com/11165551/150699804-a61735d1-aac1-46de-b0be-8d07a75893bb.jpg)

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+          </li>
+        </ol>
+      
+        <ul class="slider-list">
```
надо классы переработать. я б сказала, что класс slider-list подходит скорее для обёртки, внутри которой лежат все слайды, чем для списка характеристик товара внутри отдельного слайда. тут явно какой-то другой класс нужен. если отдельный слайд будет иметь класс slide, то картинка может иметь класс slide-image, название товара slide-title, текстовое описание slide-description, список характеристик slide-features, ссылка slide-link. как-то так

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+            </button>
+          </li>
+          <li class="slider-pagination-item">
+            <button class="slider-pagination-button" slider-pagination-button" type="radio">
```
тут и в следующей кнопке значение атрибута class разбилось лишней кавычкой. если повыпиливать вот эти кавычки, станет нормально 

![Разметка главной страницы by AlexandraPoltoratskaya · Pull Request #1 · htmlacademy-htmlcss:232847-device-33 2022-01-24 01-19-01](https://user-images.githubusercontent.com/11165551/150700226-66b6879f-1682-433a-87ca-be9cf9bd33ea.jpg)

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+        <h2 class="visually-hidden">Каталог-навигация по сайту</h2>
+        <ul class="catalog-navigation">
+          <li class="catalog-navigation-item">
+            <a class="catalog-navigation-link" href="#">
+              <img class="catalog-item-img" src="" alt="Переход на товары для виртуальной реальности">
```
если уж совсем придираться, то в описании картинки должно быть именно описание картинки, а не действие ссылки, в которой картинка лежит. то есть "иконка товаров для виртуальной реальности". "Переход на товары для виртуальной реальности" это действие ссылки, но его нет смысла дублировать скрытым текстом или aria-label, потому что текст в ссылке и так есть, и в целом понятно, что там происходит

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+        </ul>
+      </section>
+
+      <section class="benefits">
+        <h2 class="visually-hidden">Преимущества</h2>
+          <ul class="benefits-tabs-list">
+            <li class="benefit-tab active"><a class="tab-link shipping" href="#">Доставка</a></li>
+            <li class="benefit-tab"><a class="tab-link guaranty" href="#">Гарантия</a></li>
+            <li class="benefit-tab"><a class="tab-link credit" href="#">Кредит</a></li>
+          </ul>
+
+          <section class="benefit-content">
+            <h2 class="benefit-title">Доставка</h2>
+            <img class="benefit-image" src="" alt="Курьер, бегущий по лезвию">
+            <p class="benefit-message">Мы с удовольствием доставим ваш товар прямо к вашему подъезду совершенно бесплатно! Ведь мы неплохо заработаем, поднимая его на ваш этаж.</p>
+          </section>
+      </section>
```
это в каком-то смысле тоже слайдер или блок с переключающимися табами. 3 кнопки, 3 блока описаний, соответствующие кнопкам (см стайлгайд)
![image](https://user-images.githubusercontent.com/11165551/150700390-792797ed-1081-4539-98d3-2c15fe7f8db1.png)

структура примерно такая же, как в стандартном слайдере

```
сам слайдер
    блок с контролами
        кнопка
        кнопка
        кнопка
    блок со слайдами
        слайд
        слайд
        слайд

```

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+      <section class="benefits">
+        <h2 class="visually-hidden">Преимущества</h2>
+          <ul class="benefits-tabs-list">
+            <li class="benefit-tab active"><a class="tab-link shipping" href="#">Доставка</a></li>
```
это кнопки, они переключают описания. ссылки нам тут не сильно помогут, т к перехода на другую страницу тут не будет

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+          </section>
+      </section>
+      
+      <section class="delivery">
+        <h3 class="visually-hidden">Преимущество доставки</h3>
+        <img class="delivery-icon" alt="Грузовичок доставки летит к вам, как Супермен">
+        <p class="delivery-text">Доставим редкий товар под заказ</p>
+      </section>
```
это, видимо, элемент, который вызывает появление модального окна "Доставка под заказ". к тому же в стайлгайде есть для него состояния, так что он 100% интерактивный.  
![image](https://user-images.githubusercontent.com/11165551/150700525-0b90e473-0774-44df-b5b8-182eb84d8510.png)

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+        <p class="delivery-text">Доставим редкий товар под заказ</p>
+      </section>
+
+      <section class="about">
+        <h2>О нас</h2>
+        <p>Огромный выбор гаджетов не оставит равнодушным гика, который есть в каждом из нас.</p>
+        <p>Мы можем доставить ваш товар в самые отдалённые уголки России транспортными компаниями:</p>
+
+        <ul class="about-list">
+          <li class="about-item">Меловые линии</li>
+          <li class="about-item">СДИК</li>
+          <li class="about-item">ТоксБерри</li>
+        </ul>
+
+        <a class="about-link">Подробнее о нас</a>
+      </section>
+
+      <section class="contacts">
+        <h2 class="visually-hidden">Контакты</h2>
+        <p>Вы можете забрать товар сами, заехав в наш офис.<br>
+           Заодно сможете проверить работоспособность покупки.</p>
+
+        <b>Интернет-магазин «Девайс»</b>
+        <p>Санкт-Петербург, набережная реки Карповки, 5, литера П.<br>
+           +7 (812) 812-12-12</p>
+
+        <b>Время работы</b>
+        <p>пн—пт: с 10:00 до 22:00<br>
+           сб—вс: с 10:00 до 19:00</p>
+        
+        <a class="feedback-link" href="#">Напишите нам</a>
+      </section>
```
если внимательно посмотреть на эти две секции, можно увидеть, что контент в них оформлен одинаково. разве что во второй секции нет списка, а в первой ссылок. но если бы были, скорее всего тоже были бы оформлены аналогично. ну а ссылки одинаковые полностью. так что секциям и их контенту есть смысл дать одинаковые классы, чтоб не преумножать страдания 

кстати, насчёт ссылок и всего, что в макете выглядит как кнопки. в стайлгайде кнопки всех видов, использующихся на проекте, собраны в отдельный блок 
![image](https://user-images.githubusercontent.com/11165551/150700792-9fd47fb8-03d3-423c-bda0-cf79ec60b5bb.png)
 
и это стоит учитывать при именовании классов. следует исходить из того, что все одинаковые кнопки должны иметь одинаковые классы, потому что css-классы это в первую очередь про оформление. базовый класс кнопки не обязательно должен быть её единственным классом, можно добавлять дополнительные классы для описания поведения кнопки в контексте. на этапе написания стилей это станет более понятно, но пока что стоит внимательно посмотреть на стайлгайд и понять, какие элементы являются общими для всего проекта компонентами. вот кнопки точно являются

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+        <h2>Подпишитесь на рассылку</h2>
+        <p>Еженедельный дайджест с обзорами новинок, длительными тест-драйвами, распаковками, а также акции, спецпредложения и вот это вот всё.</p>
+
+        <input type="email" name="email-subscribe" placeholder="Ваш e-mail">
+        <button>Подписаться</button>
```
тут должна быть форма, а кнопка Подписаться это кнопка отправки этой формы, то есть button type="submit" https://up.htmlacademy.ru/htmlcss/33/criteries#HTML-04

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+        <input type="email" name="email-subscribe" placeholder="Ваш e-mail">
+        <button>Подписаться</button>
+
+        <img src="" alt="Галочка">
```
декоративные изображения вроде этой галочки не стоит добавлять в разметку. её можно будет сделать позже фоновой картинкой для параграфа "Это бесплатно"

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+ 
+    </main>
+    <footer class="main-footer">
+      <img class="logo-footer" src="" alt="Логотип магазина Девайс">
```
этот логотип работает так же, как в шапке, то есть является ссылкой на главную страницу

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+        </li>
+      </ul>
+
+      <p>Тел.: +7 (812) 812-12-12</p>
```
телефон надо сделать ссылкой с протоколом tel: https://up.htmlacademy.ru/htmlcss/33/criteries#HTML-03

talequale commented on Device_index.html:
```diff
@@ -0,0 +1,263 @@
+
+      <ul class="footer-social-list">
+        <li class="footer-social-item">
+          <a class="button" href="https://www.facebook.com/htmlacademy">
```
button это обычно какая-то стандартная кнопка вроде тех, что собраны в стайлгайде под заголовком "кнопки". социальная кнопка это кастомный элемент, и лучше дать ему соответствующий класс, например, social-btn / social-link

talequale reviewed (commented):
файл с разметкой главной страницы должен называться просто index.html. в рамках проекта нет необходимости в файлах уточнять название проекта

и отдельный момент - в именах файлов не должно быть прописных букв (а также пробелов, кириллических и спецсимволов)

то есть нужно было разметку главной страницы писать в том файле index.html, который в проекте уже есть

погляди внимательно стайлгайд и критерии выборочно. там есть моменты, которые в макете не очевидны

AlexandraPoltoratskaya committed:
внесла изменения в верстку 1й страницы согласно кодгайду

AlexandraPoltoratskaya committed:
внесла изменения в верстку 1й страницы

согласно стайлгайду и замечаниям наставника

talequale commented on index.html:
```diff
@@ -2,11 +2,269 @@
-    <p>Репозиторий создан для обучения на профессиональном онлайн‑курсе «<a href="https://htmlacademy.ru/intensive/htmlcss">HTML и CSS. Профессиональная вёрстка сайтов</a>».</p>
+      <nav class="user-navigation">
+        <form class="user-search-form">
+        <label for="search" class="visually-hidden">Поиск по сайту</label>
```
тут немного надо будет форматирование поправить 
![Разметка главной страницы by AlexandraPoltoratskaya · Pull Request #1 · htmlacademy-htmlcss:232847-device-33 2022-02-08 02-55-09](https://user-images.githubusercontent.com/11165551/152891946-1760306c-c0d0-4482-82e3-4c742c299d43.jpg)

видишь, label лежит внутри form, но визуально они на одном уровне, как будто это соседние теги. это мешает восприятию структуры разметки (и нарушает техническое требование соответствия кода кодгайду)

AlexandraPoltoratskaya replied:
подвинула так. правильно?

```
<form class="user-search-form">
          <label for="search" class="visually-hidden">Поиск по сайту</label>
          <input class="search"  placeholder="Текст поиска" type="search" name="search" id="search">

          <button class="button newsletter-button" type="submit">
            <span>Найти</span>
          </button>
</form>
```

talequale replied:
![image](https://user-images.githubusercontent.com/11165551/154138122-bbb4a73d-7dc5-478a-b6f4-1cc5ee17fe67.png)
тут отформатировалось странно, а в коде нормально.

есть ещё места со сбитым форматированием, например, вот 
![232847-device-33-master 2 – index html 2022-02-15 22-52-13](https://user-images.githubusercontent.com/11165551/154138414-75adb58d-e88b-43c6-b8d8-928da1a4252e.jpg)
в редакторе хорошо видно, что между закрывающими тегами ul и li на строках 59-60 расстояние в 2 отступа (в сумме 4 пробела), вместо одного отступа (2 пробела). это из-за того, что тег ul на строке 40 сдвинут вправо относительно соседнего тега button. выглядит так, как будто ul лежит внутри button, но на самом деле это не так

talequale commented on index.html:
```diff
@@ -2,11 +2,269 @@
+              <option value="fitness-bracelets">Фитнес-браслеты</option>
+              <option value="smart-watches">Умные часы</option>
+              <option value="quadrocopters">Квадрокоптеры</option>
+            </select>
```
нет, это не сортировка и не фильтр. задача сортировки - сортировать что-то по какому-то признаку. и это что-то должно присутствовать на странице. а у нас на странице ничего такого, поддающегося сортировке, нет. 
тут именно список ссылок на страницы, которые у нас есть на сайте. "каталог товаров" это не ссылка, это элемент, клик на который открывает вложенный список ссылок на страницы каталога. саму механику открытия списка нам реализовывать не нужно, но нужно подготовить разметку и стили под возможность оживления. 

структура такая: 
```
<ul class="main-navigation-list">
  <li class="main-navigation-item">
    <button class="main-navigation-link" type="button">Каталог товаров</button>
    <ul>
      <li>
        <a href="#">Виртуальная реальность</a>
      </li>
      ... дальше остальные ссылки 
    </ul>
  </li>
  <li class="main-navigation-item">
    <a class="main-navigation-link"> Доставка </a>
  </li>
  <li class="main-navigation-item">
    <a class="main-navigation-link"> Гарантия </a>
  </li>
  <li class="main-navigation-item">
    <a class="main-navigation-link"> Контакты </a>
  </li>
</ul>

```

то есть список с вложенным списком в первом пункте

talequale commented on index.html:
```diff
@@ -2,11 +2,269 @@
+        </p>
+        <a class="slider-description-link" href="#">Подробнее</a>
+
+        <ul class="slider-gallery">
```
ну это же не галерея, это переключатели слайдера. slider-controls, тогда уж. галерея предполагает скорее карусель из картинок, и в этом смысле, если что-то и является галереей, то сам слайдер. а никак не его кнопочки

и смотри, сейчас у тебя такая структура: 

```
<section class="slider-image">
  тут картинка из первого слайда и стрелки, переключающие слайды. 
  то есть по нажатию на эти стрелки мы увидим либо следующий, либо 
  предыдущий слайд. если мы скроем эту секцию  slider-image, чтобы 
  показать картинку второго слайда, то стрелки скроются вместе с ней, 
  а стрелки нам нужны независимо от того, какой слайд мы видим
</section>
<section class="slider-description">
  тут описание товара и пагинация слайдера. опять же, если мы скроем 
  этот блок slider-description, чтоб показать следующий или предыдущий, 
  то пагинация исчезнет вместе с этим блоком, а она нам нужна
</section>
```

короче, нам надо разметить не один слайд, а все три. и основной вопрос - как будет работать переключение? нам не нужно его реализовывать, но нужно подготовить структуру для того, чтобы программист мог написать логику. куда в разметке попадёт контент остальных двух слайдов? я хочу увидеть их в разметке. в прошлом ревью я писала о том, какая структура может быть у слайдера https://github.com/htmlacademy-htmlcss/232847-device-33/pull/1#discussion_r790330330 . если там что-то непонятно, напиши мне в чат, обсудим. но разметить надо весь слайдер вместе со скрытым контентом.

talequale commented on index.html:
```diff
@@ -2,11 +2,269 @@
+      <section class="slider-description">
+        <h2 class="visually-hidden">Описание квадрокоптера</h2>
+
+        <p class="slogan">Порхает как бабочка,<br>
```
тут не нужен br (и не будет нужен), я писала об этом в прошлом ревью https://github.com/htmlacademy-htmlcss/232847-device-33/pull/1#discussion_r790336752

talequale commented on index.html:
```diff
@@ -2,11 +2,269 @@
+        </ul>
+      
+        <table>
+          <tr>
+            <td>800 м</td><td>50 м</td>
+            <td>Дальность полета</td><td>Радиус поражения</td>
+          </tr>
```
tr это строка таблицы, table row. строк у нас в таблице 2, а не одна 
![HTML 1 : Девайс 2022-02-08 03-42-48](https://user-images.githubusercontent.com/11165551/152896401-817ac87d-9226-4a7e-b283-70f0feddcc9e.jpg)

а у тебя в разметке одна строка

AlexandraPoltoratskaya replied:
исправила на это
```
<table>
          <tr>
            <td>800 м</td><td>50 м</td>
          </tr>
          <tr>
            <td>Дальность полета</td><td>Радиус поражения</td>
          </tr>
        </table>
```

talequale replied:
тут норм, в таблицах в двух других слайдах строки и ячейки перепутаны. написала подробно в комментарии к коду

talequale commented on index.html:
```diff
@@ -2,11 +2,269 @@
+
+        <ul class="slider-gallery">
+          <li class="slider-gallery-item">
+            <button class="slider-gallery-button slider-gallery-button-current" type="radio">
```
не бывает `button type="radio"`, я писала об этом тут https://github.com/htmlacademy-htmlcss/232847-device-33/pull/1#discussion_r790337051

AlexandraPoltoratskaya replied:
а! значит, я должна исправить с button на input, потому что у второго есть тип radio. верно?

talequale replied:
нет, не верно. нужно исправить значение атрибута type. нам не нужен input type="radio", это же не форма, не работа с данными. это просто кнопка, у которой предполагается какая-то логика, которая будет написана программистом

talequale commented on index.html:
```diff
@@ -2,11 +2,269 @@
+            <h2 class="benefit-title">Доставка</h2>
+            <p class="benefit-message">Мы с удовольствием доставим ваш товар прямо к вашему подъезду совершенно бесплатно! Ведь мы неплохо заработаем, поднимая его на ваш этаж.</p>
+          </section>
+      </section>
```
тут тот же вопрос, что и к первому слайдеру - нужна разметка остальных слайдов. они показаны в стайлгайде https://github.com/htmlacademy-htmlcss/232847-device-33/pull/1#discussion_r790339298

talequale commented on index.html:
```diff
@@ -2,11 +2,269 @@
+          </ul>
+
+          <section class="benefit-content">
+            <h2 class="benefit-title">Доставка</h2>
```
если следовать иерархии блоков, тут должен быть заголовок h3, так как в родительском блоке заголовок имеет уровень h2

talequale commented on index.html:
```diff
@@ -2,11 +2,269 @@
+      <section class="delivery">
+        <h3 class="visually-hidden">Преимущество доставки</h3>
+        <p class="delivery-text">Доставим редкий товар под заказ</p>
+      </section>
```
тут не вижу правок по вот этому комментарию. https://github.com/htmlacademy-htmlcss/232847-device-33/pull/1#discussion_r790339989 
элемент интерактивный - значит это что-то, с чем пользователь может взаимодействовать. это может быть ссылка, кнопка, поле формы. в нашем случае подходит кнопка, потому что элемент не должен вызывать перехода на другую страницу (то есть это не ссылка) и не должен собирать и отправлять данные (то есть не поле формы)

talequale commented on index.html:
```diff
@@ -2,11 +2,269 @@
+        <p class="delivery-text">Доставим редкий товар под заказ</p>
+      </section>
+
+      <section class="about">
```
https://github.com/htmlacademy-htmlcss/232847-device-33/pull/1#discussion_r790340939 тут тоже не вижу изменений

talequale reviewed (commented):
не по всем комментариям увидела исправления. если что-то непонятно, или ты с чем-то не согласна, пиши, пожалуйста, мне, разберёмся и обсудим. но я всё-таки пишу комментарии не просто потому, что хочу высказаться. как правило пишу в случаях, когда объективно требуются изменения, и ожидаю реакции

AlexandraPoltoratskaya committed:
внесены изменения в верстку слайдера

AlexandraPoltoratskaya committed:
изменения в слайдере

AlexandraPoltoratskaya committed:
внесла исправления

сделала блок "Доставка" интерактивным
переназвала классы для ссылок "Подробнее"

talequale commented on index.html:
```diff
@@ -2,11 +2,354 @@
+      <h1 class="visually-hidden">Девайс - интернет-магазин гаджетов</h1>
+
+      <section class="slider">
+        <h2 class="visually-hidden">Слайдер</h2>
+        
+          <div class="slider-image-quadrocopter">
+            <h3 class="visually-hidden">Изображение квадрокоптера на слайдере</h3>
+            <img class="slider-quadrocopter-big-image" src="" alt="Квадрокоптер.">
+          </div>
+
```
![Разметка главной страницы by AlexandraPoltoratskaya · Pull Request #1 · htmlacademy-htmlcss:232847-device-33 2022-02-15 22-56-30](https://user-images.githubusercontent.com/11165551/154139897-0c1418a8-7b94-4d0d-8638-99c527508fd9.jpg)
смотри, тут тоже код выглядит так, как будто див с классом slider-image-quadrocopter лежит в скрытом заголовке h2. погугли, как в твоём редакторе делается автоформатирование кода. если это **vs code**, то для Windows Shift + Alt + F
для Mac Shift + Option + F
для Linux Ctrl + Shift + I
если **webstorm**, то 
![Без названия 2022-02-15 23-05-16](https://user-images.githubusercontent.com/11165551/154140387-ca31b84c-fccb-433e-bafc-d69e9f2887ef.jpg)

AlexandraPoltoratskaya replied:
поняла, сделала vs code Windows Shift + Alt + F

talequale commented on index.html:
```diff
@@ -2,11 +2,354 @@
+            </li>
+          </ul>
+
+      </section>
```
разметка слайдера стала лучше, но ещё не совсем правильно.
из хорошего: ты выделила контролы (стрелки и круглые переключатели) в отдельные блоки, и теперь они не дублируются в каждом слайде 

из не очень пока удачного: 
- контент слайдов всё ещё по сути свален в кучу в блоке slider, это сделает логику переключения слайдов сложной и замысловатой без необходимости
- отдельные части слайдов имеют разные классы, хотя части слайдов имеют одинаковое оформление. грубо говоря, у нас три слайда, стало быть, три блока с текстовым описанием (slider-description-quadrocopter, slider-description-fitness-braselet, slider-description-monopodes-for-selfie). это значит, что ты трижды напишешь по сути одни и те же стили для каждого из блоков. потому что элемент один, а классы разные

смотри, я жду чего-то такого: https://jsfiddle.net/theano/zdwo6n45/29/
![gllacy slider - JSFiddle - Code Playground 2022-02-15 23-57-25](https://user-images.githubusercontent.com/11165551/154148022-d2112ef0-5c18-4bb5-89db-1021c38e16bf.jpg)
логика переключения слайдов проста: мы прячем все слайды, показываем слайд, у которого в данный момент есть класс active. скрипт переносит этот класс со слайда на слайд в зависимости от того, на какую кнопку нажал пользователь. 
то есть у нас есть обёртка slide, в которой лежит весь контент слайда (картинка, описание, ссылка, что угодно). эта обёртка отвечает за видимость слайда (если у неё есть дополнительный класс active, слайд виден. если нет - не виден). и собственно для этого она нам и нужна, потому что иначе скрипту надо будет работать не с одним элементом, а с разрозненной кучей контента разных слайдов. (и также эта обёртка понадобится нам для сеток, но это потом) 

в среднем примерно по такому принципу работают карусели

talequale commented on index.html:
```diff
@@ -2,11 +2,354 @@
+
+          <table class="table-quadrocopter">
+            <tr>
+              <td>800 м</td><td>50 м</td>
```
лучше делать перенос строки между тегами, чтоб было видно, что тут два тега td, а не один. читаемость кода это важно

talequale commented on index.html:
```diff
@@ -2,11 +2,354 @@
+            <a class="link-about" href="#">Подробнее</a>
+          </div>
+
+          <table>
+            <tr>
+              <td>48 часов</td><td>Без подзарядки</td>
+            </tr>
+          </table>        
```
tr это строка. строки в таблицах описания характеристик в этом слайдере всегда две. ячеек в строках может быть разное количество 
![HTML 1 : Девайс 2022-02-16 00-02-27](https://user-images.githubusercontent.com/11165551/154148864-51e83aad-a7e8-4d47-86bd-0655d41f262b.jpg)

я вижу тут 2 возможные причины затруднений:
1) недостаточно хорошо усвоена теория, нет чёткого представления о том, как работают и ведут себя таблицы
2) с толку сбивает то, что по сути тут не табличные данные, они просто представлены в похожем на таблицу виде. поэтому не очевидно, что вот есть строка с заголовками, а есть строка с данными, соответствующими этим заголовкам. можно тут вообще не использовать тег table, а представить этот блок как список характеристик товара:

```
<ul class="product-features">
  <li class="product-feature">
    <p class="product-feature-value">48 часов</p> 
    <p class="product-feature-label">Без подзарядки</p>   
  </li>
  <li class="product-feature">
    <p class="product-feature-value">800 м</p> 
    <p class="product-feature-label">Дальность полета</p>   
  </li>
</ul>
```

с точки зрения семантики это будет правильнее. и тут можно группировать характеристики логически, а не "графически", как требует таблица

talequale commented on index.html:
```diff
@@ -2,11 +2,354 @@
+            <a class="link-about" href="#">Подробнее</a>
+          </div>
+  
+          <table>
+            <tr>
+              <td>8,5 м</td><td>Длина палки</td>
+            </tr>
+            <tr>
+              <td>5 кг</td><td>Вес палки</td>
+            </tr>
+            <tr>
+              <td>Карбон</td><td>Материал</td>
+            </tr>
+          </table>              
```
тут та же история, что и в прошлой таблице, перепутаны местами строки и ячейки

talequale commented on index.html:
```diff
@@ -2,11 +2,354 @@
+
+          <ul class="slider-controls">
+            <li class="slider-controls-item">
+              <button class="slider-controls slider-controls-current" type="button">
```
класс slider-controls уже использован на списке кнопок. дублирование его на отдельной кнопке ни к чему хорошему не приведёт, ведь это совсем разные элементы, и одинаковые стили им ни к чему. к тому же slider-controls это множественное число, "контролы слайдера". списку подходит, кнопке не подходит. кнопке можно дать класс slider-control ("контрол слайдера", ед.ч.)

AlexandraPoltoratskaya replied:
то есть, должно быть так, верно? ul -- это controlS, а li и button -- это control

```
<ul class="slider-controls">
        <li class="slider-control-item">
          <button class="slider-control slider-control-current" type="button">
            <span class="visually-hidden">Квадрокоптер</span>
          </button>
        </li>
```

talequale replied:
да, всё так

talequale commented on index.html:
```diff
@@ -2,11 +2,354 @@
+        </ul>
+      </section>
+
+      <section class="benefits-slider">
+        <h2 class="visually-hidden">Преимущества</h2>
+
+        <ul class="benefits-controls">
+          <li class="benefits-controls-item">
+            <button class="tab-link tab-link-shipping" type="button">Доставка</button>
+          </li>
+          <li class="benefits-controls-item">
+            <button class="tab-link tab-link-guarantee" type="button">Гарантия</button>
+          </li>
+          <li class="benefits-controls-item">
+            <button class="tab-link tab-link-credit" type="button">Кредит</button>
+          </li>
+        </ul>
+
+        <div class="benefits-description">
+
+          <div class="benefits-shipping-description">
+            <p class="benefits-shipping-description">Доставка</p>
+            <p>Мы с удовольствием доставим ваш товар прямо к вашему подъезду совершенно бесплатно! Ведь мы неплохо заработаем, поднимая его на ваш этаж.</p>
+          </div>
+        
+          <div class="benefits-quarantee-description">
+            <p class="benefits-quarantee-description">Гарантия</p>
+            <p>Если из-за возгорания купленного у нас товара у вас сгорит дом — не переживайте, мы выдадим вам новый. Товар, не дом, конечно же.</p>
+          </div>
+
+          <div class="benefits-credit-description">
+            <p class="benefits-credit-description">Кредит</p>
+            <p>Залезть в долговую яму стало проще! Кредитные консультанты подберут для вас наиболее выгодные условия кредита. Выгодные для банка, разумеется.</p>
+          </div>
+
+        </div>
+
+      </section>
```
вот у этого слайда хорошая понятная структура. только немного классы слишком разнообразные, тут набор типовых элементов 
![Разметка главной страницы by AlexandraPoltoratskaya · Pull Request #1 · htmlacademy-htmlcss:232847-device-33 2022-02-16 00-18-24](https://user-images.githubusercontent.com/11165551/154151202-5918ac45-81ae-4d45-9fa5-3c414f665a51.jpg)

talequale commented on index.html:
```diff
@@ -2,11 +2,354 @@
+        <a class="link-about" href="#">Напишите нам</a>
+      </section>
+
+      <section class="newsletter">
+        <h2>Подпишитесь на рассылку</h2>
+        <p>Еженедельный дайджест с обзорами новинок, длительными тест-драйвами, распаковками, а также акции, спецпредложения и вот это вот всё.</p>
+
+        <form class="newsletter-form" action="https://echo.htmlacademy.ru/" method="post">
+
+          <label for="newsletter-email" class="visually-hidden">Email</label>
+          <input class="field"  placeholder="Ваш e-mail" type="email" name="newsletter-email" id="newsletter-email" required>
+
+          <button class="button newsletter-button" type="submit">
+            <span>Подписаться</span>
+          </button>
+
+        </form>
+
+        <p class="newsletter-free">Это бесплатно</p>
+      </section>      
```
модальные окна обычно пишут в самом конце body, то есть его надо вытащить из main и поместить после футера

AlexandraPoltoratskaya replied:
то есть, выглядит так?

```
    <a href="https://htmlacademy.ru/"><img class="logo-academy" src="" alt="Логотип HTML Academy."></a>

  </footer>

  <form class="newsletter-form" action="https://echo.htmlacademy.ru/" method="post">

    <label for="newsletter-email" class="visually-hidden">Email</label>
    <input class="field" placeholder="Ваш e-mail" type="email" name="newsletter-email" id="newsletter-email"
      required>

    <button class="button newsletter-button" type="submit">
      <span>Подписаться</span>
    </button>

  </form>
  
</body>
```

talequale replied:
ой, не, сорри, это я всё напутала 😵‍💫 я перепутала форму подписки вот с этим модальным окном. оно должно будет оказаться после футера 
![image](https://user-images.githubusercontent.com/11165551/154621253-070c1ada-cc8b-4554-af13-613942bb3601.png)

а форму конечно нужно вернуть в секцию newsletter. извини ещё раз!

AlexandraPoltoratskaya replied:
++ вернула
эээ, а как пишутся модальные окна?

вот пример с htmlbook.ru, все верно?

```
<title>Событие onsubmit</title>
  <script>
   function deleteName(f) {
    if (confirm("Вы уверены, что хотите удалить выделенный пункт?\nЭта операция не восстановима.")) 
       f.submit();
   }
  </script>
 </head>
 <body>
  <form action="form.php" onsubmit="deleteName(this);return false;">
   <p>Выберите пункт для удаления</p>
   <p><input type="radio" name="p" value="1"> Чебурашка</p>
   <p><input type="radio" name="p" value="2"> Крокодил Гена</p>
   <p><input type="radio" name="p" value="3"> Шапокляк</p>
   <p><input type="submit" value="Удалить"></p>
  </form>
```

talequale commented on index.html:
```diff
@@ -2,11 +2,354 @@
+ 
+    </main>
+    <footer class="main-footer">
+      <a href="#"><img class="logo-footer" src="" alt="Логотип магазина Девайс."></a>
```
класс лучше вешать на внешний элемент (ссылку), а не на внутренний (картинку). потому что от класса внешнего элемента мы по каскаду можем добраться до внутреннего, а наоборот, от внутреннего к внешнему, - не можем

AlexandraPoltoratskaya replied:
`  <a class="logo-footer" href="#"> <img src="" alt="Логотип магазина Девайс."></a>` верно?

talequale replied:
да

AlexandraPoltoratskaya committed:
изменения в слайдере

AlexandraPoltoratskaya committed:
изменения во 2 слайдере и футере

talequale commented on index.html:
```diff
@@ -1,12 +1,372 @@
+        <p class="description">Самая длинная палка для селфи доступна в нашем магазине. Восемь (Восемь, Карл!) метров
+          длиной и весом всего 5 кг.</p>
+        <a class="link-about" href="#">Подробнее</a>
+      </div>
```
тут что-то пошло не так с классами, нужно сделать аналогично первым двум слайдам

talequale commented on index.html:
```diff
@@ -1,12 +1,372 @@
+          <p class="product-feature-label">Материал</p>   
+        </li>
+      </ul>
+
```
в целом правильно, но не хватает обёрток для слайдов
![Разметка главной страницы by AlexandraPoltoratskaya · Pull Request #1 · htmlacademy-htmlcss:232847-device-33 2022-02-18 07-55-45](https://user-images.githubusercontent.com/11165551/154620259-74485c58-f456-4813-ae1c-9e1740e45151.jpg)
типа такого. у нас в слайдере 3 слайда, то есть 3 блока slider-item. в каждом из которых лежат 1 slider-image, 1 slider-description, 1 product-features

AlexandraPoltoratskaya replied:
++ исправила

talequale commented on index.html:
```diff
@@ -1,12 +1,372 @@
+
+      <div class="benefits-description">
+
+        <div class="benefit-description-item">
```
![Разметка главной страницы by AlexandraPoltoratskaya · Pull Request #1 · htmlacademy-htmlcss:232847-device-33 2022-02-18 08-01-15](https://user-images.githubusercontent.com/11165551/154620779-c8c0133a-930b-4ec4-858a-028efc107d65.jpg)

AlexandraPoltoratskaya replied:
++

talequale commented on index.html:
```diff
@@ -1,12 +1,372 @@
+      </li>
+    </ul>
+
+    <a href="https://htmlacademy.ru/"><img class="logo-academy" src="" alt="Логотип HTML Academy."></a>
```
тут, как и в случае с логотипом, класс logo-academy лучше повесить на ссылку, а не на картинку

AlexandraPoltoratskaya replied:
++

AlexandraPoltoratskaya committed:
добавила обертку div слайдов

talequale commented on index.html:
```diff
@@ -1,12 +1,380 @@
+        </li>
+      </ul>
+
+      <div class="benefits-description">
+
+        <div class="benefit-description">
+          <p class="benefit-description-title">Доставка</p>
+          <p>Мы с удовольствием доставим ваш товар прямо к вашему подъезду совершенно бесплатно! Ведь мы неплохо
+            заработаем, поднимая его на ваш этаж.</p>
+        </div>
+
+        <div class="benefit-description">
+          <p class="benefit-description-title">Гарантия</p>
+          <p>Если из-за возгорания купленного у нас товара у вас сгорит дом — не переживайте, мы выдадим вам новый.
+            Товар, не дом, конечно же.</p>
+        </div>
+
+        <div class="benefit-description">
+          <p class="benefit-description-title">Кредит</p>
+          <p>Залезть в долговую яму стало проще! Кредитные консультанты подберут для вас наиболее выгодные условия
+            кредита. Выгодные для банка, разумеется.</p>
+        </div>
+
+      </div>
```
класс блока не может быть таким же, как классы его детей 
![Разметка главной страницы by AlexandraPoltoratskaya · Pull Request #1 · htmlacademy-htmlcss:232847-device-33 2022-02-20 01-39-23](https://user-images.githubusercontent.com/11165551/154821432-86818a18-aadd-45bc-8c3e-92dadbcd36cc.jpg)
надо либо одно, либо другое слегка изменить

AlexandraPoltoratskaya replied:
сделала так:
```
<div class="benefits-list">

        <div class="benefit-description">
```
